src/main/java/tasks/TaskManager.java

```diff
@@ -0,0 +1,224 @@
+package tasks;
+
+import java.util.List;
+import java.util.regex.Pattern;
+
+import exceptions.DukeCommandException;
+import exceptions.DukeDateTimeException;
+import exceptions.DukeIndexException;
+import exceptions.DukeIoException;
+import exceptions.DukeNoInputException;
+
+
+/**
+ * TaskManager is a class to handle where tasks are CRUD.
+ */
+public class TaskManager {
+    private final List<Task> taskList;
+    private final TaskIoParser ioParser;
+    private final TextParser textParser;
+
+    /**
+     * Constructs TaskManager for the Duke Application with loading from the save file
+     * @param path File Path from Main Class
+     * @throws DukeIoException If no loaded save is read
+     */
+    public TaskManager(String path) throws DukeIoException {
+        this.ioParser = new TaskIoParser(path);
+        this.taskList = ioParser.loadTaskList();
+        this.textParser = new TextParser();
```

Comment:
Referring to the JDoc for all rules (the one suggested for W3 was only basic + intermediate), the `this` keyword is best used "only when a field is shadowed by a method or constructor parameter".

In this case, since `path` does not shadow any of the `TaskManager` attributes (`taskList`, `ioParser`, and `textParser`), perhaps it is safe and good practice to remove the `this` keyword.

Reply:
:0 oh noes im used to the python way of using this/self at all times. Thanks for alerting me to this. Sorry for the late response to your review!